tests/e2e/specs/add-edit-category.test.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	visitAdminPage,
+	pressKeyWithModifier,
+	pressKeyTimes
+} from '@wordpress/e2e-test-utils'
+import { addQueryArgs } from '@wordpress/url';
+
+describe( 'Categories tests', () => {
+	const title = 'New Category';
+	const query = addQueryArgs( '', {
+		taxonomy: 'category',
+	} );
+
+	beforeEach( async () => {
+		/**
+		 * Delete all categories before anything
+		 * This is useful because after running more than one test
+		 * there could be existing categories
+		 */
+		await visitAdminPage( 'edit-tags.php', query );
+		await page.$( '#bulk-action-selector-top' );
+		await page.waitForSelector( '[id^=cb-select-all-]' );
+		await page.click( '[id^=cb-select-all-]' );
+		await page.select( '#bulk-action-selector-top', 'delete' );
+		await page.click( '#doaction' );
+
+		/**
+		 * Create a new category with the title 'New Category'
+		 */
+		await page.waitForSelector('#tag-name');
+		await page.focus( '#tag-name' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await page.type( '#tag-name', title );
+		await page.click( '#submit' );
+
+		await page.reload();
```

Comment:
Do we need to reload the page after creating the new category?

Reply:
Yes, I've noticed that when you add a new category, the new one is not added to `document.querySelectorAll('#the-list tr')` node. So the node's length doesn't change. You have to reload the page before the new one is counted.

Could this be a bug in WordPress?

Reply:
Hmm, that's more of a timing issue with the test: when, say, in the first test, the line `const categories = await page.$$( '#the-list tr' );` runs, the categories list hasn't been updated yet. Reloading the page makes sure it is updated, but we could also use something like `await page.waitForSelector('#the-list tr + tr')` (or `tr:nth-child(x)`, depending on how many rows we expect there to be) to check the table has updated before  going on with the test. 

The main advantage of avoiding reloading the page is speed, as the reload will probably take longer than just waiting for a selector to display on the page.